resteasy-jaxrs/src/main/java/org/jboss/resteasy/core/ResourceMethodRegistry.java

```diff
@@ -279,6 +280,45 @@ protected void register(ResourceFactory rf, String base, ResourceClass resourceC
       }
    }
 
+   /**
+    * Resteasy 2.x does not properly handle sub-resource and sub-resource locator
+    * endpoints with the same uri.  Resteasy 3.x does handle this properly.  In
+    * assisting customers identify this issue during an upgrade from Resteasy 2 to 3
+    * provides a waring when the situation is found.
+    */
+   public void checkAmbiguousUri()
+   {
+      for (Map.Entry<String, List<ResourceInvoker>> entry : this.root.getBounded().entrySet())
+      {
+         List<ResourceInvoker> values = entry.getValue();
+         if (values.size() > 1) {
+            int locatorCnt = 0;
+            int methodCnt = 0;
+            for(ResourceInvoker rInvoker : values)
+            {
+               if (rInvoker instanceof ResourceLocatorInvoker)
+               {
+                  locatorCnt++;
+               } else if (rInvoker instanceof ResourceMethodInvoker)
+               {
+                  methodCnt++;
+               }
+            }
+            if (methodCnt > 0 && locatorCnt > 0)
+            {
+               StringBuilder sb = new StringBuilder();
+               for (ResourceInvoker exp : values) {
+                  sb.append(exp.getMethod().getDeclaringClass().getName() + "."
+                          + exp.getMethod().getName()+ ", ");
+               }
+
+               LogMessages.LOGGER.uriAmbiguity(entry.getKey(),
+                       sb.toString().substring(0, sb.toString().length()-2));
+            }

```

Comment:
This is also improvable; instead of creating yet a new String to crop the last 2 chars, we should only add the ", " before if not dealing with the last ResourceInvoker value. So please refactor the loop above instead. (btw StringBuilder also have a length() method, which does not requiring building the string)